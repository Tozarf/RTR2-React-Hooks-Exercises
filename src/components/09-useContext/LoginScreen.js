import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const {setUser,user} = useContext(UserContext)

    console.log(user)
    return (
        <>
            <h1>Login Screen</h1>
            <hr/>
            <button
                className="btn btn-primary"
                onClick={()=>setUser({
                    id:1909,
                    name: "Fausto"
                })}
            
            >
                Login
            </button>
        </>
    )
}
