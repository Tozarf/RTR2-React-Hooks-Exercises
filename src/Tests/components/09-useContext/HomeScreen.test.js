import React from 'react'
import { mount } from 'enzyme'
import { UserContext } from '../../../components/09-useContext/UserContext'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen'




describe('testing <HomeScreen/>', () => {
    const user = {
        name : "Fausto",
        email : "user@example.com"
    }
    const wrapper = mount(
    <UserContext.Provider value = {{
        user
    }}>
        <HomeScreen/>
    </UserContext.Provider>
    )

    test('should match snapshot ', () => {

        expect(wrapper).toMatchSnapshot()

    })
    
})
