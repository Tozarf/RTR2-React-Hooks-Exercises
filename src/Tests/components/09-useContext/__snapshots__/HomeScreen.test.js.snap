// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <HomeScreen/> should match snapshot  1`] = `
Array [
  <h1>
    Home Screen
  </h1>,
  <hr />,
  <pre>
    {
   "name": "Fausto",
   "email": "user@example.com"
}
  </pre>,
]
`;
